src/xunit.performance.api/SafeTerminateHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Xunit.Performance.Api.Native.Windows;
+using System;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal sealed class SafeTerminateHandler<T> : IDisposable
```

Comment:
I wanted to make clear through the name that it's meant to handle terminate signals (currently only on Windows). Therefore, I think `SafeDisposableWrapper` would omit the fact that it disposes when a terminate signal is received by the application.
Thoughts?